### PR TITLE
Replace output to stdout/stderr with Log calls and fix code annotations

### PR DESCRIPTION
**Pull request by thom-at-redhat**

### CODE REVIEW

File: pkg/routes/imagesets.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func ListAllImageSets(w http.ResponseWriter, r *http.Request) {
 		json.NewEncoder(w).Encode(&countErr)
 		return
 	}
-	fmt.Printf("r.URL.Query() %v \n", r.URL.Query().Get("sort_by"))
+	log.Errorf("r.URL.Query() %v \n", r.URL.Query().Get("sort_by"))
```
Comment (biancarosa) [inline pkg/routes/imagesets.go:118]:
```suggestion
	log.Debugf("r.URL.Query() %v \n", r.URL.Query().Get("sort_by"))
```

maybe? doesn't look like an error or something relevant enough for Info level

File: pkg/services/images.go [hunk 2]
```diff
@@ -287,17 +287,17 @@ func (s *ImageService) CreateRepoForImage(i *models.Image) *models.Repo {
 	}
 	tx := db.DB.Create(repo)
 	db.DB.Save(&repo)
-	fmt.Printf("Repo:: %d\n", repo.ID)
-	fmt.Printf("i.commit:: %d\n", i.Commit.ID)
+	log.Infof("Repo:: %d\n", repo.ID)
```
Comment (biancarosa) [inline pkg/services/images.go:290]:
```suggestion
	log.Debugf("Repo:: %d\n", repo.ID)
```

File: pkg/services/images.go [hunk 3]
```diff
@@ -287,17 +287,17 @@ func (s *ImageService) CreateRepoForImage(i *models.Image) *models.Repo {
...
+	log.Infof("i.commit:: %d\n", i.Commit.ID)
```
Comment (biancarosa) [inline pkg/services/images.go:291]:
```suggestion
	log.Debugf("i.commit:: %d\n", i.Commit.ID)
```

File: pkg/services/images.go [hunk 4]
```diff
@@ -287,17 +287,17 @@ func (s *ImageService) CreateRepoForImage(i *models.Image) *models.Repo {
...
 	i.Commit.RepoID = &repo.ID
 
 	tx2 := db.DB.Save(i.Commit)
 	if tx2.Error != nil {
-		fmt.Printf("::TX2:: %v\n", tx2.Error)
+		log.Infof("::TX2:: %v\n", tx2.Error)
 		panic(tx2.Error)
 	}
-	fmt.Printf("i.commit:: %d\n", i.Commit.RepoID)
+	log.Infof("i.commit:: %d\n", i.Commit.RepoID)
```
Comment (biancarosa) [inline pkg/services/images.go:300]:
```suggestion
	log.Debugf("i.commit:: %d\n", i.Commit.RepoID)
```

File: pkg/services/images.go [hunk 5]
```diff
@@ -287,17 +287,17 @@ func (s *ImageService) CreateRepoForImage(i *models.Image) *models.Repo {
...
-	fmt.Printf("i.commit:: %d\n", i.Commit.ID)
+	log.Infof("Repo:: %d\n", repo.ID)
+	log.Infof("i.commit:: %d\n", i.Commit.ID)
 	i.Commit.Repo = repo
 	i.Commit.RepoID = &repo.ID
 
 	tx2 := db.DB.Save(i.Commit)
 	if tx2.Error != nil {
-		fmt.Printf("::TX2:: %v\n", tx2.Error)
+		log.Infof("::TX2:: %v\n", tx2.Error)
```
Comment (biancarosa) [inline pkg/services/images.go:297]:
```suggestion
		log.Errorf("::TX2:: %v\n", tx2.Error)
```

There's a panic down there and this is our last chance to log this in error level, as far as I can see.

File: pkg/services/updates.go [hunk 6]
```diff
@@ -223,7 +223,7 @@ func (s *UpdateService) RebootDevice(update *models.UpdateTransaction) {
 		filePath := cfg.TemplatesPath
 		templateName := "template_playbook_dispatcher_reboot_device.yml"
 		template, err := template.ParseFiles(filePath + templateName)
-		fmt.Printf("template: %v", template)
+		log.Infof("template: %v", template)
```
Comment (biancarosa) [inline pkg/services/updates.go:226]:
```suggestion
		log.Debugf("template: %v", template)
```

Also debug here? Looks useful for development but not so much for prod.

**Review by biancarosa — APPROVED** [review]
LGTM! And thanks for fixing the other comments, too 👍🏻

**Comment by thom-at-redhat** [comment]
Also resolves https://issues.redhat.com/browse/THEEDGE-1230

**Pull request merged**